src/sgame/sg_buildable.cpp

```diff
@@ -1702,6 +1702,40 @@ itemBuildError_t G_CanBuild( gentity_t *ent, buildable_t buildable, int /*distan
 		level.numBuildablesForRemoval = 0;
 	}
 
+	int max_miners = g_maxMiners.Get();
+	if (max_miners >= 0)
+	{
+		// maybe rewrite loop as
+		//   ForEntities([](Entity& entity, MiningComponent& mining) {});
+		gentity_t *tmp = g_entities + MAX_CLIENTS;
+		int miners = 0;
+		for (int i = MAX_CLIENTS; i < level.num_entities; i++, tmp++) 
+		{
+			if (tmp->s.eType == entityType_t::ET_BUILDABLE) 
+			{
+				HealthComponent* health = tmp->entity->Get<HealthComponent>();
+				if (!G_OnSameTeam(tmp, ent) || !health->Alive() || !Entities::IsAlive(tmp))
```

Comment:
```suggestion
				if (!G_OnSameTeam(tmp, ent) || !health->Alive() || !Entities::IsAlive(tmp))
```
```suggestion
				if (!G_OnSameTeam(tmp, ent) || !Entities::IsAlive(tmp))
```
and you can remove the line above. IsAlive uses the healthcomponent and doesn't crash if it's missing

Reply:
I removed that.